app/Http/Controllers/DeviceGroupController.php

```diff
@@ -25,11 +26,32 @@ public function index()
     {
         $this->authorize('manage', DeviceGroup::class);
 
+        $device_groups = DeviceGroup::orderBy('name')->withCount('devices')->get();
```

Comment:
This is not very optimized, you could _probably_ add `->with('devices:device_id')`  
I'm not sure if `pluck()` supports relationships, else you could do something like:
```php
foreach($device_groups as $device_group {
  $device_ids[] = $device_group->devices->pluck('device_id');
}
```

Reply:
i tried it with 'pluck' but i didn't get it to run

Reply:
Seems `pluck()` doesn't support relationships.  
But your current code does N+1 queries, if you do the `->with('devices:device_id')`, it will [eager load it](https://laravel.com/docs/5.8/eloquent-relationships#eager-loading).